filament/src/main/java/net/fabricmc/filament/FilamentGradlePlugin.java

```diff
@@ -60,25 +62,22 @@ public void apply(Project project) {
 		tasks.register("generatePackageInfoMappings", GeneratePackageInfoMappingsTask.class);
 		tasks.register("javadocLint", JavadocLintTask.class);
 
-		TaskProvider<CombineUnpickDefinitionsTask> combineUnpickDefinitions = tasks.register("combineUnpickDefinitions", CombineUnpickDefinitionsTask.class);
+		var combineUnpickDefinitions = tasks.register("combineUnpickDefinitions", CombineUnpickDefinitionsTask.class);
 		tasks.register("remapUnpickDefinitionsIntermediary", RemapUnpickDefinitionsTask.class, task -> {
 			task.dependsOn(combineUnpickDefinitions);
 			task.getInput().set(combineUnpickDefinitions.flatMap(CombineUnpickDefinitionsTask::getOutput));
 			task.getSourceNamespace().set("named");
 			task.getTargetNamespace().set("intermediary");
 		});
 
-		project.getConfigurations().register("minecraftLibraries", configuration -> configuration.setTransitive(false));
+		tasks.register("minecraftLibraries", MinecraftLibrariesTask.class, task -> {
+			task.dependsOn(minecraftMetaTask);
 
-		GradleUtils.afterSuccessfulEvaluation(project, () -> afterEvaluate(project));
-	}
-
-	private void afterEvaluate(Project project) {
-		try {
-			MinecraftMetadata.setup(project);
-		} catch (Exception e) {
-			throw new RuntimeException("Failed to setup", e);
-		}
+			var files = MinecraftVersionMetaTask.readMetadata(minecraftMetaTask)
+					.map(meta -> MinecraftLibrariesTask.getDependencies(project, meta))
+					.map(dependencies -> project.getConfigurations().detachedConfiguration(dependencies).resolve());
+			task.getFiles().from(files);
```

Comment:
This is pretty clever lol, the task itself does nothing

Reply:
Took me quite a few attempts to get this to work, shame I couldnt make a task output work to populate a normal configurations inputs.